test/es-module/test-esm-loader-search.js

```diff
@@ -18,3 +18,12 @@ assert.throws(
     message: /Cannot find package 'target'/
   }
 );
+
+assert.throws(
+  () => resolve('fs/promisesx'),
+  {
+    code: 'ERR_MODULE_NOT_FOUND',
+    name: 'Error',
+    message: /Cannot find package 'fs\/promisesx'/
```

Comment:
I don't think this is correct, the missing package is `fs`, not `fs/promisesx`. What you could do is to add the complete specifier – either as `Cannot find package 'fs' imported from … while trying to resolve 'fs/promisesx'`, or as an additional property on the error.

Reply:
IMO for the end-users, `Cannot find package 'fs'` will not hint them that `fs/promisesx` is the not-found package/specifier.

But I'm not an expert on this, so IDK.

Reply:
+1 to the suggestion by @aduh95 as this giving more context.

For builtins, it might even be worth having a different error message, that is builtin specific:

`Unable to import builtin module 'fs/promisesx'`.

And if we're doing wishful thinking, would be amazing to see levenshtein distance searching:

`Unable to import builtin module 'fs/promisesx', did you mean 'fs/promises'`?.

Such suggestion error messages could in theory even apply to node_modules packages lookups and exports as well.

Reply:
Suggestion error messages seem like a good idea to have in general :-)